File: array-reverse/js/main_builtin.js
if (process.argv.length !== 4) {
  console.error('Usage: node main.js <size> <iterations>');
  process.exit(1);
}

var size = parseInt(process.argv[2]);
var iterations = parseInt(process.argv[3]);

// generate the array
list = []
for (var i = 0; i < size; i++) {
  list.push(i);
}

for (var i = 0; i < iterations; i++) {
  list.reverse();
}

File: array-reverse/rust/main_builtin.rs
use std::env::args;

fn main() {
  if args().count() != 3 {
    panic!("Usage: {} <size> <iterations>", args().next().unwrap());
  }

  let mut args_iter = args().skip(1);

  let count: usize = args_iter.next().unwrap().parse().unwrap();
  let iterations: usize = args_iter.next().unwrap().parse().unwrap();

  let mut array: Vec<usize> = (0..count).collect();

  for _ in 0..iterations {
    array.reverse();
  }
}
